source/base/mpi.cc

```diff
@@ -247,6 +247,7 @@ namespace Utilities
       AssertThrow(ierr == MPI_SUCCESS, ExcInternalError());
 
       ierr = MPI_Type_commit(&type);
+      AssertThrow(ierr == MPI_SUCCESS, ExcInternalError());
```

Comment:
Oh, this is interesting. I suspect that the number of MPI calls for which we check the return code is very small. We should probably do a better job about this.